Code/Framework/AzToolsFramework/AzToolsFramework/API/EditorAssetSystemAPI.h

```diff
@@ -309,21 +259,16 @@ namespace AzToolsFramework
             virtual AZ::Outcome<AZStd::string> GetJobLog(AZ::u64 jobrunkey) = 0;
         };
 
-        inline const char* GetHostAssetPlatform()
-        {
-#if defined(AZ_PLATFORM_MAC)
-            return "mac";
-#elif defined(AZ_PLATFORM_WINDOWS)
-            return "pc";
-#elif defined(AZ_PLATFORM_LINUX)
-            return "linux";
-#else
-            #error Unimplemented Host Asset Platform
-#endif
-        }
+        //! Returns "mac", "pc", or "linux" statically.
+        extern const char* GetHostAssetPlatform();
```

Comment:
This can be moved to PlatformDefaults.h.
Also a function is external by default, so the `extern` keyword isn't needed for a function declaration.